File: app/src/main/java/com/example/kunal/shophelper/Acitivites/SplashActivity.java
package com.example.kunal.shophelper.Acitivites;

import android.content.Intent;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;

import com.example.kunal.shophelper.HelperClasses.Constants;
import com.example.kunal.shophelper.R;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        runhandler();
    }

    private void runhandler() {

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
//                isphonenumbersaved();
                if (!isphonenumbersaved()) {
                    Intent i = new Intent(SplashActivity.this, LoginPreview.class);
                    startActivity(i);
                    finish();
                } else {
                    Intent i = new Intent(SplashActivity.this, LoginActivity.class);
                    startActivity(i);
                    finish();
                }
            }
        }, 3000);
    }

    private boolean isphonenumbersaved() {
        if (!Constants.getstoredata(Constants.PhoneNumber, "0", this).equals("0")) {
//            Toast.makeText(this, ""+Constants.getstoredata(Constants.PhoneNumber,"0",this), Toast.LENGTH_SHORT).show();
            return true;
        } else {
//            Toast.makeText(this, "null", Toast.LENGTH_SHORT).show();
            return false;
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
//        runhandler();
    }
}

File: app/src/main/java/com/example/kunal/shophelper/Splash/SplashActivity.kt
package com.example.kunal.shophelper.Splash

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import com.example.kunal.shophelper.Login.LoginActivity
import com.example.kunal.shophelper.R
import com.example.kunal.shophelper.ShopHelperApplication
import com.example.kunal.shophelper.Splash.Presenter.Splashpresenter
import com.example.kunal.shophelper.Splash.View.SplashView

class SplashActivity:AppCompatActivity(),SplashView {
    lateinit var mypresenter:Splashpresenter<SplashView>
    override fun startintent(intent: Intent) {
        startActivity(intent)
        finish()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        var dataManager=(application as ShopHelperApplication).getDataManager()
        mypresenter= Splashpresenter(dataManager,this)
        mypresenter.onattach(this)
        mypresenter.runhandler(1000)
    }

}